internal/testrunner/coverageoutput.go

```diff
@@ -42,11 +48,19 @@ func (tcd *testCoverageDetails) withTestResults(results []TestResult) *testCover
 			tcd.dataStreams[result.DataStream] = []string{}
 		}
 		tcd.dataStreams[result.DataStream] = append(tcd.dataStreams[result.DataStream], result.Name)
+		if tcd.cobertura != nil && result.Coverage != nil {
+			if err := tcd.cobertura.Merge(result.Coverage); err != nil {
+				tcd.errors = append(tcd.errors, errors.Wrapf(err, "can't merge Cobertura coverage for test `%s`", result.Name))
+			}
+		} else if tcd.cobertura == nil {
+			tcd.cobertura = result.Coverage
+		}
 	}
 	return tcd
 }
 
-type coberturaCoverage struct {
+// CoberturaCoverage is the root element for a Cobertura XML report.
+type CoberturaCoverage struct {
```

Comment:
Is there any reason why these properties are exposed? Is it for the consistency with `CoberturaCoverage`?

Reply:
I think I got mixed here a little bit. The original idea was for the test runners to have the option to generate their own coverage in Cobertura format (via the new TestResult.Coverage), that's why these types are exposed.

To align with this idea, I think it makes more sense to put GetPipelineCoverage into the pipeline runner instead of the global testrunner package as is now.

Reply:
> To align with this idea, I think it makes more sense to put GetPipelineCoverage into the pipeline runner instead of the global testrunner package as is now.

Yes, I have a similar feeling about this :)